src/metabase/driver/postgres.clj

```diff
@@ -296,6 +300,57 @@
   [driver [_ arg]]
   (sql.qp/->honeysql driver [:percentile arg 0.5]))
 
+(defmethod sql.qp/->honeysql [:postgres :datetimediff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)
+        helper
+        (fn helper [x y unit]
+          (case unit
+            (:year :day)
+            (hx/cast
+             :integer
+             (hsql/call
+              :date_part
+              (hx/literal (name unit))
+              (hsql/call
+               (case unit :year :age :day :-)
+               (hsql/call :date_trunc (hsql/raw "'day'") y)
```

Comment:
```suggestion
               (hsql/call :date_trunc (hx/literal :day)  y)
```
I think this will work, we used this in https://github.com/metabase/metabase/blob/deep-copy-dashboards/src/metabase/driver/postgres.clj#L243

Reply:
It would work, but I question the value of adding more indirection here. We're already writing in raw SQL with `hsql/call :date_trunc` etc, so the closer everything is to raw SQL the more consistent it reads IMO.

Reply:
Using helpers like `hx/+`, `hx/floor` is helpful because the indirection actually makes the code read more similar to raw SQL, but `hx/literal` doesn't achieve that.

Reply:
This is more or less a matter of taste, but IMO looking at sth with a keyword is more readable, compare to `"'day'"`, the combination of double-quote and single-quote is a bit verbose I think :D.

Reply:
Yeah, either taste or my lack of familiarity. I'll leave it as is for now. I actually prefer the double quotation marks